packages/govuk-frontend/src/govuk/errors/index.mjs

```diff
@@ -47,3 +47,28 @@ export class SupportError extends GOVUKFrontendError {
 export class ConfigError extends GOVUKFrontendError {
   name = 'ConfigError'
 }
+
+/**
+ * Indicates an issue with an element (possibly `null` or `undefined`)
+ */
+export class ElementError extends GOVUKFrontendError {
+  name = 'ElementError'
+
+  /**
+   * @param {Element} element - The element in error
+   * @param {object} options - options
+   * @param {string} options.componentName - The name of the component throwing the error
+   * @param {string} options.identifier - An identifier that'll let the user understand which element has an error (variable name, CSS selector)
+   * @param {typeof HTMLElement} [options.expectedType] - The type that was expected for the element
+   */
+  constructor(
+    element,
+    { componentName, identifier, expectedType = HTMLElement }
+  ) {
+    const reason = !element
+      ? `${identifier} not found`
+      : `${identifier} is not an instance of "${expectedType.name}"`
```

Comment:
Prettier really isn't helping these parameter lines 🙈

Reminds me of a recent discussion on destructuring with default values too

Think you and @36degrees would prefer this?

```suggestion
  constructor(element, options) {
    const expectedType = options.expectedType || HTMLElement
    
    const reason = !element
      ? `${options.identifier} not found`
      : `${options.identifier} is not an instance of "${expectedType.name}"`
```

Reply:
I'll update after lunch 😊

Reply:
Whilst [fixing those conflicts](https://github.com/alphagov/govuk-frontend/pull/4104#issuecomment-1723436569) for you I've pushed up a suggestion that keeps your original destructuring with:

```js
  constructor(element, { componentName, identifier, expectedType }) {
    expectedType = options.expectedType || window.HTMLElement
```

But we have Babel now [kept another branch](https://github.com/alphagov/govuk-frontend/compare/components-type-error...components-type-error-nullish#diff-e17a722e3c0cbe12ed4c41bcb10900a1cd5b8d13c0e10825e222baaedd046bdb) if you'd prefer:

```js
  constructor(element, { componentName, identifier, expectedType }) {
    expectedType ??= window.HTMLElement
```